### DISCUSSION TITLE
Cannot Connect to Postgres

### DISCUSSION THREAD

**Discussion by mertturkmenoglu** [nextauthjs/next-auth#1973] [Help] [answered]
Hello everyone,

I am trying to use the email provider and Postgres database. I enter my email, an email magic link comes to my mailbox. After I click the link, I get the following result:

`TypeError: Cannot read property 'isConnected' of null`

![Screenshot 2021-05-11 161327](https://user-images.githubusercontent.com/34170799/117821420-3f20d980-b274-11eb-9fab-6f8950bd5833.png)

![Screenshot 2021-05-11 162928](https://user-images.githubusercontent.com/34170799/117823406-1ac5fc80-b276-11eb-9c0f-bfbd2b9d29c6.png)

I inspect the database and the tables are there. I can see the verification_requests table and its data.

I tried using different databases and different configurations (via database URI, or TypeORM config object). The result is the same. 

Thank you.

**Comment by balazsorban44** [comment, marked as answer]
Hi! This has been reported over at #1968

I pushed a fix for this for the TypeORM adapter, and if you run `npm i` or `yarn` and make sure that your `package-lock.json` or `yarn.lock` lists at least `@next-auth/typeorm-legacy-adapter@0.0.2-canary.129`, it should be resolved for now! Feel free to give further feedback on the above-linked issue, if that is still not the case! Thank you!

**Reply by mertturkmenoglu** [reply]
Hi,

Running `npm i` didn't do it but I removed `node_modules` folder and `package-lock.json` file and tried to run the same command, and this time it worked.

Before trying this solution, I found a workaround but I'm not sure if this is how it is supposed to be:

```typescript
import NextAuth from 'next-auth';
import Providers from 'next-auth/providers';
import Adapters from 'next-auth/adapters';

export default NextAuth({
  providers: [
    // Providers
  ],
  adapter: Adapters.TypeORM.Adapter({
     type: 'postgres',
     host: 'localhost',
     port: 5432,
     username: 'myusername',
     password: 'mypassword',
     database: 'mydatabase',
     synchronize: true,
  }),
});
```

I couldn't found an example of this, I was trying random things with the help of IntelliSense. But this code did the job for me.

Thank you.

**Reply by balazsorban44** [reply]
looks good!
